github/login.go

```diff
@@ -83,3 +92,19 @@ func validateResponse(user *github.User, resp *github.Response, err error) error
 	}
 	return nil
 }
+
+func githubClientFromAuthURL(authURL string, httpClient *http.Client) (*github.Client, error) {
```

Comment:
This should be table testable, you can compare the Client's BaseURL with an expected one.